features/126-record-id/api.md

```diff
@@ -0,0 +1,73 @@
+# Removing `type/` from APIs
+
+## Overview of changes
+
+This document propose what changes are needed to remove `type/` from Skygear API
+so that developer will have consistent experience when needing to specify
+a Record ID.
+
+Throughout this document the different ID formats are described as:
+
+Deprecated ID - "note/0ebd484c-ed82-4b04-aba7-33fc530f3c88"
+Record Type - "note"
+Record ID - "0ebd484c-ed82-4b04-aba7-33fc530f3c88"
+
+### Changes to SDK API
+
+* API that accepts deprecated ID as parameter should accept a Record Type and
+  a Record ID as parameter, both of `string` type.
+* API that accepts a list of deprecated IDs as parameter should accept a Record
+  Type (string) and a list of Record ID (string).
+* API that returns deprecated IDs (or list) should return Record ID (string),
```

Comment:
similarly, have we got an overview of our api, that when shall we return object or ID?

Reply:
That depends on the API?

fetchRecord: ID > Record
saveRecord: Record > Record
deleteRecord: (ID or Record) > ID (returning an object after deleting it doesn’t seem intuitive)